### DISCUSSION TITLE
Make fixed "showing unsaved changes" bar

### DISCUSSION THREAD

**Discussion by gioppe** [craftcms/cms#11586] [Ideas] [answered]
Just a visual tweak idea: when the "unsaved changes" bar appears it pushes down the whole layout. If you are already editing and clicking around it's easy to mis-click something, lose focus and other annoyances. It would be helpful if things remained where they appear :)

**Comment by thupsi** [comment]
There is an issue about this (https://github.com/craftcms/cms/issues/11229) and I guess it was kinda resolved with a muted click event, but yeah, I also would prefer the bar not to cause a UI shift. 

Brandon says that the shift is intentional, but when the user has scrolled the shift doesn't happen PLUS the bar is not visible. 

I'm not sure about the optimal solution. Making the bar fixed would cover some useful UI on the top. Maybe making the bar fixed to the bottom would be a good compromise? As for making sure it's noticed, perhaps there could be a subtle animation, or some other visual device.

**Reply by brandonkelly** [reply]
We actually have an idea on how to resolve this for Craft 4.2 – will update this and #11229 if it works out :)

**Comment by brandonkelly** [comment, marked as answer]
This is now resolved for the next Craft 4 release (6229cc3635b6fefe9a091bde50def57dc76f5383). Going forward, if there’s enough content on the page to require a scrollbar, the page will automatically scroll itself down by 50 pixels (the height of the notice) to compensate, so the fields won’t budge, visually.

When there’s not enough content for a scrollbar, the page will continue to jump down like before.

**Reply by brandonkelly** [reply]
Craft 4.2.0 is out now with that change.
